metrics/ci_integrations/test/source/buildkite/config/validator/buildkite_source_validator_test.dart

```diff
@@ -0,0 +1,556 @@
+      () {
+        final validator = BuildkiteSourceValidator(
+          validationDelegate,
+          validationResultBuilder,
+        );
+
+        expect(validator.validationDelegate, equals(validationDelegate));
+        expect(
+          validator.validationResultBuilder,
+          equals(validationResultBuilder),
+        );
+      },
+    );
+
+    test(
+      ".validate() delegates the access token validation to the validation delegate",
+      () {
+        whenValidateAuth().thenSuccessWith(noOrganizationScopeToken);
+
+        final expectedAuth = BearerAuthorization(accessToken);
+
+        validator.validate(config);
+
+        verify(validationDelegate.validateAuth(expectedAuth)).called(1);
+      },
+    );
+
+    test(
+      ".validate() returns a validation result with the successful access token field validation result if the access token is valid",
+      () async {
+        whenValidateAuth().thenSuccessWith(noOrganizationScopeToken);
+
+        final result = await validator.validate(config);
+
+        final validationResult = _getAuthValidationResult(result);
+
+        expect(validationResult, MatcherUtil.isSuccessfulValidationResult);
+      },
+    );
+
+    test(
+      ".validate() returns a validation result with the access token field validation result, having the additional context equal to the message returned by the validation delegate if the access token is valid",
+      () async {
+        whenValidateAuth().thenSuccessWith(noOrganizationScopeToken, message);
+
+        final result = await validator.validate(config);
+
+        final validationResult = _getAuthValidationResult(result);
+        final additionalContext = validationResult.additionalContext;
+
+        expect(additionalContext, equals(message));
+      },
+    );
+
+    test(
+      ".validate() returns a validation result with the failure access token field validation result if the access token is invalid",
+      () async {
+        whenValidateAuth().thenErrorWith();
+
+        final result = await validator.validate(config);
+
+        final validationResult = _getAuthValidationResult(result);
+
+        expect(validationResult, MatcherUtil.isFailureValidationResult);
+      },
+    );
+
+    test(
+      ".validate() returns a validation result with the access token field validation result, having the additional context equal to the message returned by the validation delegate if the access token is invalid",
+      () async {
+        whenValidateAuth().thenErrorWith(null, message);
+
+        final result = await validator.validate(config);
+
+        final validationResult = _getAuthValidationResult(result);
+        final additionalContext = validationResult.additionalContext;
+
+        expect(additionalContext, equals(message));
+      },
+    );
+
+    test(
+      ".validate() returns a validation result with the unknown organization slug field validation result if the access token is invalid",
```

Comment:
Would it be better to test that the validate method calls the `setEmptyResults` method after auth token validation failed?